federation/pkg/federatedtypes/adapter.go

```diff
@@ -53,9 +55,35 @@ type FederatedTypeAdapter interface {
 	ClusterUpdate(client kubeclientset.Interface, obj pkgruntime.Object) (pkgruntime.Object, error)
 	ClusterWatch(client kubeclientset.Interface, namespace string, options metav1.ListOptions) (watch.Interface, error)
 
+	IsSchedulingAdapter() bool
+
 	NewTestObject(namespace string) pkgruntime.Object
 }
 
+// SchedulingStatus contains the status of the objects that are being
+// scheduled into joined clusters.
+type SchedulingStatus struct {
```

Comment:
Please update these to generic types (or make them interfaces, where they are passed/used).